app/views/core/CreateAccountModal/BasicInfoView.coffee

```diff
@@ -75,7 +80,7 @@ module.exports = class BasicInfoView extends CocoView
   checkEmail: ->
     email = @$('[name="email"]').val()
 
-    if @hideEmail
+    if isCodeCombat and @hideEmail
```

Comment:
with above change, you might not need it. But you will have better idea after making change